app/models/post.rb

```diff
@@ -332,4 +338,34 @@ def invalidate_caches
     return unless saved_change_to_authors_locked?
     Post::Author.clear_cache_for(authors)
   end
+
+  def create_written
+    return if skip_written
+    self.written = Reply.create!(
+      post: self,
+      reply_order: 0,
+      user: user,
+      content: content,
+      icon: icon,
+      character: character,
+      character_alias: character_alias,
+      created_at: created_at,
+      updated_at: edited_at,
+      skip_regenerate: true,
+    )
+  end
+
+  WRITTEN_ATTRS = %w(content icon_id character_alias_id character_id)
+
+  def update_written
+    return unless written.present?
+    return if (changed_attributes.keys - WRITTEN_ATTRS).empty?
+    written.update!(
+      content: content,
+      icon: icon,
+      character: character,
+      character_alias: character_alias,
+      updated_at: edited_at,
```

Comment:
Do we need `skip_post_update` here, too? It looks like it'd be re-triggering this post update if not, recursively.